src/spec/create_extension_spec.py

```diff
@@ -0,0 +1,531 @@
+                    "wiring for an example: "
+                    "https://probeinterface.readthedocs.io/en/main/examples/ex_11_automatic_wiring.html#sphx-glr-examples-ex-11-automatic-wiring-py"
+                ),
+                dtype="int",
+                quantity="?",
+            ),
+            NWBDatasetSpec(
+                name="shank_id",
+                neurodata_type_inc="VectorData",
+                doc="Shank ID of the contact",
+                dtype="text",
+                quantity="?",
+            ),
+            NWBDatasetSpec(
+                name="plane_axes",
+                neurodata_type_inc="VectorData",
+                doc=(
+                    "The axes defining the contact plane for each contact. It can be used for contact-wise rotations. "
+                    "For 2D probes, provide two points (v1x, v1y) and (v2x, v2y). Axis 1 is defined by the vector "
+                    "from (0, 0) to (v1x, v1y). Axis 2 is defined by the vector from (0, 0) to (v2x, v2y). "
+                    "So for one contact, a 45 degree rotation would be "
+                    "[[1 / sqrt(2), 1 / sqrt(2)], [-1 / sqrt(2), 1 / sqrt(2)]]. "
+                    "The default is [[1, 0], [0, 1]]. For 3D probes, provide two points (v1x, v1y, v1z), "
+                    "(v2x, v2y, v2z). "
+                    "See 'contact_plane_axes' in "
+                    "https://probeinterface.readthedocs.io/en/main/format_spec.html for more details."
+                ),
+                dtype="float",
+                dims=[["num_contacts", "v1, v2", "x, y"], ["num_contacts", "v1, v2", "x, y, z"]],
+                shape=[[None, 2, 2], [None, 2, 3]],
+                quantity="?",
+            ),
+            NWBDatasetSpec(
+                name="shape",
+                neurodata_type_inc="VectorData",
+                doc="Shape of the contact; e.g. 'circle'",
+                dtype="text",
+                quantity="?",
+            ),
+            NWBDatasetSpec(
+                name="radius_in_um",
+                neurodata_type_inc="VectorData",
+                doc="Radius of a circular contact, in micrometers.",
+                dtype="float",
+                quantity="?",
+            ),
+            NWBDatasetSpec(
+                name="width_in_um",
+                neurodata_type_inc="VectorData",
+                doc="Width of a rectangular or square contact, in micrometers.",
+                dtype="float",
+                quantity="?",
+            ),
+            NWBDatasetSpec(
+                name="height_in_um",
+                neurodata_type_inc="VectorData",
+                doc="Height of a rectangular contact, in micrometers.",
+                dtype="float",
+                quantity="?",
+            ),
+        ],
+    )
+
+    probe = NWBGroupSpec(
+        neurodata_type_def="Probe",
+        neurodata_type_inc="Device",
```

Comment:
There was an interesting debate in the in-person discussion about making a distinction between Devices that would be something similar to a class (the model that you can look at a catalogue of a manufacturer) and instances of them that would be the actual physical sensors on the experimental arrangment. 

It made sense to me that the ProbeModel would be a Device but it was unclear what type the (instance) Probe would be.

This probably is out of the scope here and I don't see any downside about defining both as subclasses of Device but I was still curios to hear your thoughts about it. Do we gain anything by making such distinction?

Reply:
In NWB currently, the two concepts for Device are mixed together. If someone has two Neuropixels probes, they will create two Device objects, one for each probe, with different names. This Device has a `manufacturer` that are really a property of the model, not the instance. 

I think by making such a distinction, we reduce redundancy and improve interoperability with libraries like probeinterface and any catalogs of microscopes or other devices. It would be useful to have a `DeviceModel` type in core NWB. That's a bigger change for NWB that I think is beyond the scope of this extension.

Reply:
Cool. This makes sense to me. 
> It would be useful to have a DeviceModel type in core NWB. That's a bigger change for NWB that I think is beyond the scope of this extension.

Agree this is out of scope. Thanks for sharing your thoughts. I also think this distinction  makes sense in Probeinterface. I will keep thinking about it.